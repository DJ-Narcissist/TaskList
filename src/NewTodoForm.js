import React, { useState} from "react";
import { v4 as uuid } from "uuid";
import Todo from "./TodoList";

const NewTodoForm = () => {
    const [formData, setFormData] = useState({

    });

    
    

    const gatherInput = (e) => {
        e.preventDefault();
        addItem({ ...formData, id: uuid() });
        setFormData({item:"", value:""})
    };



    return (
        <form onSubmit={gatherInput}>
            <label htmlFor="key">
                Task:
                <input
                type="text"
                name={task}
                value={formData.key}
                onChange={handleChange}
                />

            </label>
           
        </form>
        
    );
}

export default NewTodoForm;