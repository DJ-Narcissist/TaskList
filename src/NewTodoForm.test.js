import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", () => {
    render (<NewTodoForm />);
});

it("matches the snapshot",() => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on submit", () => {
    const createMock = jest.fn();
    const { getByText } = render(<NewTodoForm addItem={createMock} />);
    const createButton = getByText("Add Task");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});
