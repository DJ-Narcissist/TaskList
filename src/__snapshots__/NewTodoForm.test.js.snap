// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="key"
    >
      Task:
      <input
        name="task"
        type="text"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
